splice_machine/src/main/java/com/splicemachine/derby/procedures/SpliceAdmin.java

```diff
@@ -1718,6 +1724,72 @@ public static void SYSCS_GET_ACTIVE_SESSIONS(ResultSet[] resultSet) throws SQLEx
         resultSet[0] = res.getResultSet();
     }
 
+    public static void SYSCS_DROP_OBSOLETE_TEMPORARY_TABLES(String schemaName, final ResultSet[] resultSet) throws SQLException {
+        ResultHelper res = new ResultHelper();
+        ResultHelper.VarcharColumn tblNameCol = res.addVarchar("tableName", 128);
+        try (EmbedConnection conn = (EmbedConnection) getDefaultConn()) {
+            try {
+                if (schemaName == null) {
+                    throw StandardException.newException(LANG_INVALID_FUNCTION_ARGUMENT, "null", "SYSCS_DROP_OBSOLETE_TEMPORARY_TABLES");
+                }
+
+                schemaName = EngineUtils.validateSchema(schemaName);
+
+                LanguageConnectionContext lcc = conn.getLanguageConnection();
+                DataDictionary dd = lcc.getDataDictionary();
+                dd.startWriting(lcc);
+
+                // Get table descriptor list first, never change order!
+                SchemaDescriptor sd = getSchemaDescriptor(schemaName, lcc, dd);
+                List<TableDescriptor> tds = StatisticsProcedures.getAllTableDescriptors(sd, conn);
+                if (tds.isEmpty()) {
+                    return;
+                }
+
+                // Then get active session list.
+                List<String> activeSessionIDs = SIDriver.driver().getSessionsWatcher().getAllActiveSessions();
+                Set<UUID> sessionIDSet = activeSessionIDs.stream().map(UUID::fromString).collect(Collectors.toSet());
+
+                for (TableDescriptor td : tds) {
+                    if (td.getTableType() == TableDescriptor.LOCAL_TEMPORARY_TABLE_TYPE) {
+                        String tableName = td.getName();
+                        try {
+                            java.util.UUID creatingSessionID = lcc.getLocalTempTableSessionID(td);
+                            if (!sessionIDSet.contains(creatingSessionID)) {
+                                dropTempTable(lcc, td);
+                                res.newRow();
+                                tblNameCol.set(tableName);
+                            }
+                        } catch (StandardException se) {
+                            if (se.getSQLState().equals(SQLState.LANG_INVALID_INTERNAL_TEMP_TABLE_NAME)) {
+                                // for temporary tables created using instance numbers
+                                dropTempTable(lcc, td);
+                                res.newRow();
```

Comment:
why execute same logic in try and catch block?

Reply:
Good catch, logic is revised in the latest commit into a `finally` block.